src/pytorch_lightning/trainer/trainer.py

```diff
@@ -1894,6 +1894,14 @@ def sanity_checking(self, val: bool) -> None:
         elif self.sanity_checking:
             self.state.stage = None
 
+    @property
+    def received_sigterm(self) -> bool:
+        """Whether a SIGTERM signal was received (in at least one process)
+
+        For example, this can be checked to exit gracefully.
+        """
+        return self.strategy.reduce_boolean_decision(self._received_sigterm, all=False)
```

Comment:
Should we init this attribute as False in the Trainer's init?
How is it even possible mypy doesn't complain here?

Reply:
It is defined in the SignalConnector.

mypy does not understand self attributes defined outside the module. This is one issue caused by using connectors.